### PR TITLE
Adding PSM Gas Consumption script

### PR DESCRIPTION
**Pull request by ben-chain**
## Description
Adds a script to our tests that allows you to check out the PSM cost of `ovmSSTORE` and `ovmSLOAD` under various conditions.
## Questions
- should this be put somewhere else?

### Fixes
- Partially addresses YAS-423

## Contributing Agreement

- [x] I have read and understood the [Optimism Contributing Guide and Code of Conduct](https://github.com/ethereum-optimism/optimism-monorepo/blob/master/.github/CONTRIBUTING.md) and am following those guidelines in this pull request.

### CODE REVIEW

File: packages/contracts/contracts/test-helpers/SimpleStorage.sol [hunk 1]
```diff
@@ -4,10 +4,44 @@ contract SimpleStorage {
     mapping(bytes32 => bytes32) public builtInStorage;
 
     function setStorage(bytes32 key, bytes32 value) public {
-        builtInStorage[key] = value;
+        bytes memory EMcalldata = abi.encodeWithSelector(bytes4(keccak256(bytes("ovmSSTORE()"))), key, value);
```
Comment (kfichter) [inline packages/contracts/contracts/test-helpers/SimpleStorage.sol:7]: Can we rename this to `SimpleStorageOVM` just to clarify the separation?

File: packages/contracts/test/contracts/ovm/PartialStateManager.gas-metering-script.spec.ts [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+  .map((v, i) => {
+    return numberToHexString(startIndexForMultipleKeys + i, 32)
+  })
+const val = '0x' + '23'.repeat(32)
+
+/*********
+ * TESTS *
+ *********/
+
+describe.skip('Partial State Manager -- Storage Performance Testing Script', () => {
```
Comment (kfichter) [inline packages/contracts/test/contracts/ovm/PartialStateManager.gas-metering-script.spec.ts:48]: Any reason why this is skipped right now?

File: packages/contracts/test/contracts/ovm/PartialStateManager.gas-metering-script.spec.ts [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+    for (const aKey of multipleSequentialKeys) {
+      await stateManager.insertVerifiedStorage(
+        simpleStorageOVMAddress,
+        aKey,
+        '0x' + '00'.repeat(32)
+      )
+    }
+  })
+
+  it('setStorage', async () => {
```
Comment (kfichter) [inline packages/contracts/test/contracts/ovm/PartialStateManager.gas-metering-script.spec.ts:188]: I think generally better to put each function into a separate `describe()` and then list out the cases within an `it()`.

**Comment by K-Ho** [comment]
Closing in favor of https://github.com/ethereum-optimism/contracts-v2/pull/71

**Pull request closed without merging**